### ISSUE TITLE
[ENHANCEMENT] Global Batch Load Balancing for MoE Models

### ISSUE THREAD

**Issue by Taishi-N324** [NVIDIA/Megatron-LM#1498]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Implement global-batch level load balancing.

<img width="785" alt="Image" src="https://github.com/user-attachments/assets/970fbd9d-d26d-49ff-b871-117e8452b297" />
[Demons in the Detail: On Implementing Load Balancing Loss for Training Specialized Mixture-of-Expert Models](https://arxiv.org/abs/2501.11873)

**Benefits**
Based on the paper 

- Improves pre-training perplexity by ~0.1
- Increases benchmark scores by ~2 points
- Enables interpretable domain specialization of experts

**Comment by yanring** [comment]
Thanks! Let us take a deeper look @Victarry

**Comment by Victarry** [comment]
We will add this feature to MCore v0.13. The ETA is end of this month.

**Comment by Taishi-N324** [comment]
Thank you for reviewing and accepting this feature request! I greatly appreciate your comprehensive support and the ongoing development and maintenance of Megatron-LM.

**Comment by sbhavani** [comment]
Finished with [72d23540d0358ae24a41ff289d1461b094a770fa](https://github.com/NVIDIA/Megatron-LM/commit/72d23540d0358ae24a41ff289d1461b094a770fa)

**Comment by Taishi-N324** [comment]
Thanks very much, I’ll try it out right away!
